### DISCUSSION TITLE
What does the Idle condition duration do while "If not"?

### DISCUSSION THREAD

**Discussion by dpcollier128** [WarmUpTill/SceneSwitcher#1201] [Q&A] [answered]
It seems pretty simple, but I'm curious how the Idle condition duration is factored in and haven't found a clear answer yet. Is it satisfied as an "exactly," "at least," or "at most" duration condition? Also, with "not" applied to the Idle condition, how does the duration behavior change?

I'm trying to make a macro that automatically pauses and resumes recording based on keyboard and mouse activity.

**Comment by WarmUpTill** [comment, marked as answer]
I assume you are are using Windows:

If that is the case, the condition uses the [getlastinputinfo](https://learn.microsoft.com/bs-latn-ba/windows/win32/api/winuser/nf-winuser-getlastinputinfo) Windows API call.
This gives information about "when the last input event was received."
This value is subtracted from the current time.
If the result of this operation exceeds the given duration the condition will return true and otherwise false.

So, this a long way of saying it uses the "at least" type of duration check.

The idle condition itself does not know, what type of logical operator is applied to it.
So when applying a negation its returned value will simply be negated, which will result in the condition behaving as "if there was any input with the the last X seconds".

I hope this clears things up!
If you require me to make any changes to support your use case let me know.
